src/MetadataManager.actor.cpp

```diff
@@ -148,69 +146,71 @@ constructContext(Namespace ns, Reference<DocTransaction> tr, DocumentLayer* docL
 		    tr->tr, {StringRef(ns.first), StringRef(ns.second), StringRef(DocLayerConstants::METADATA)}));
 		state Reference<UnboundCollectionContext> tcx =
 		    Reference<UnboundCollectionContext>(new UnboundCollectionContext(tcollectionDirectory, tmetadataDirectory));
-		// fprintf(stderr, "%s.%s Creating: Collection dir: %s Metadata dir:%s Caller:%s\n", dbName.c_str(),
-		// collectionName.c_str(), printable(tcollectionDirectory->key()).c_str(),
-		// printable(tmetadataDirectory->key()).c_str(), "");
+
 		tcx->bindCollectionContext(tr)->bumpMetadataVersion(); // We start at version 1.
 		TraceEvent(SevInfo, "BumpMetadataVersion")
 		    .detail("reason", "createCollection")
 		    .detail("ns", fullCollNameToString(ns));
 
-		return std::make_pair(tcx, -1); // So we don't pollute the cache in case this transaction never commits
+		return tcx;
```

Comment:
It seems like in this case the context will never be cached since it has the version -1. It's not a problem since later when there is a query comes in, we can always read out the current version, then create and cache the context correctly, I just want to confirm this is (not cache the context when the collection got newly created) a done on purpose.

Reply:
We never cache context of a newly created collection as the transaction is not committed yet. It’s possible transaction might fail to commit. We only ever cache the context when it’s read from the database.